stan/math/prim/arr/fun/promote_elements.hpp

```diff
@@ -3,6 +3,7 @@
 
 #include <stan/math/prim/scal/fun/promote_elements.hpp>
 #include <vector>
+#include <cstddef>
```

Comment:
No, that's fine --- if you need it for `size_t` --- that always gets included for me in `<vector>` so we don't usually need it.